### PR TITLE
Add get_records() to API and df() to Registry

### LINKED ISSUE #43: Add conversion to `data.frame` to `Registry` class
Something equivalent to `ln.Artifact.df()`

### PR DESCRIPTION
**Pull request by lazappi**
Adds a `get_records()` method to the `InstanceAPI` class which performs the API call to retrieve record summary information and a `df()` method to the `Registry` class which makes the required API calls to get the requested number of records and formats the returned information as a `data.frame()`. This tries to be semi-smart to limit the number of API calls.

Fixes #43 

Checks:

- [x] CI runs
- [ ] Update `CHANGELOG`

### CODE REVIEW

File: R/Registry.R [hunk 1]
```diff
@@ -70,6 +70,73 @@ Registry <- R6::R6Class( # nolint object_name_linter
       private$.record_class$new(data = data)
     },
     #' @description
+    #' Get a data frame summarising records in the registry
+    #'
+    #' @param limit Maximum number of records to return
+    #' @param verbose Boolean, whether to print progress messages
+    #'
+    #' @return A data.frame containing the available records
```
Comment (lazappi) [inline R/Registry.R:78]: This might be nicer as a `tibble` but would need an extra dependency

File: R/Registry.R [hunk 2]
```diff
@@ -70,6 +70,73 @@ Registry <- R6::R6Class( # nolint object_name_linter
...
+      data_list <- list()
+      attr(data_list, "finished") <- FALSE
+      data_list <- purrr::reduce(
+        cli::cli_progress_along(seq_len(n_requests), name = "Sending requests"),
+        \(.data_list, .n) {
+          # Hacky way of avoiding unneeded requests until there is an easy way
+          # to get the total number of records
+          if (isTRUE(attr(.data_list, "finished"))) {
+            return(.data_list)
+          }
```
Comment (lazappi) [inline R/Registry.R:98]: If there is a way to get the total number of records we can use a regular `purrr::map` instead

**Review by rcannood — CHANGES_REQUESTED** [review]
Nice!

Could you add a unit test for df()?

File: R/Registry.R (hunk 1 above)
Comment (lazappi) [reply R/Registry.R:78]: I checked but surprisingly only through suggested dependencies at the moment

**Review by rcannood — APPROVED** [review]
LGTM!

**Pull request merged**